plasma_framework/docs/design/exit_bond_and_bounty_mechanics.md

```diff
@@ -0,0 +1,196 @@
+# Exit Bond and Bounty Mechanics
+
+This document is a description of the current implementation of exit bonds and process exit bounties which are an integral part of the exit mechanism.
+### Terminology
+
+- **Exit Initiator** The user who starts an exit by either calling `startStandardExit()` or piggybacking on an in-flight exit
+- **Challenger** The user who challenges a standard exit, an in-flight exit or a piggyback
+- **Exit Processor** The user who calls `processExit()`
+- **Exit Bond** The bond put up by the **Exit Initiator**
+- **Process Exit Bounty** A portion of the **Exit Bond** that is given to the **Exit Processor** as a reward for processing the exit.
+Whenever an exit is started, the **Exit Initiator** must put up an **Exit Bond**.
+If the exit is proven to be invalid the entire **Exit Bond** is awarded to the **Challenger**.
+If the exit is valid then after the challenge period it can be **processed** i.e. the funds returned to the **Exit Initiator**. A portion of the **Exit Bond**, called the **Process Exit Bounty**, is awarded to the **Exit Processor**. The remainder of the **Exit Bond** is returned to the **Exit Initiator**.
+Note that the **Exit Initiator** can also be the **Exit Processor**.
+
+## The Bond Life Cycle
+
+### For Standard Exits
+
+The **Exit Bond** is supplied by the **Exit Initiator** when starting an exit. Its main motive is to disincentivize invalid exits, but in the case of a valid exit a portion of it is used to incentivize processing the exit by rewarding the **Exit Processor**.
+The **Exit Initiator** should obtain the correct size of the Exit Bond by calling `startStandardExitBondSize()`.
+
+**_After Starting an Exit:_**
+*Bond -> from `Exit Initiator` to `Exit Game Contract`*
+
+For an invalid exit, a challenge transfers the whole bond to the challenger. Non-exitable exits still remain in the exit queue and are deleted when the exits are processed. For a scenario where there are mutliple invalid exits in the queue, the processor is still awarded with some gas reward for deleting non-exitable exits from storage.
+
+**_On succesfull challenge:_**
+*Bond -> from `Exit Game Contract` to `Challenger`*
+
+The **Exit Processor** gets the bounty as a reward for processing an exit. The cost of processing multiple exits from the exit queue at once is less than the cost of processing them individually. For simplicity and to save gas however, the **Process Exit Bounty** amount associated with each exit is given out to the **Exit Processor**.
```

Comment:
> For simplicity and to save gas however,

I am a bit lost on what this part is aiming to explain actually 😅 where it is simplifying and saving gas?

Reply:
Thanks! I think it will be better to point out the stacking exits by the `exit processor` for more incentives part here instead, since saving gas is on their(the exit processors) side.

Reply:
updated `ddcb22f`